core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/OAbstractPaginatedStorage.java

```diff
@@ -1337,6 +1337,93 @@ public boolean isDeleted(ORID rid) {
     }
   }
 
+  public Iterator<OBrowsePage> browseCluster(int clusterId) {
+    try {
+      checkOpenness();
+      stateLock.acquireReadLock();
+      try {
+        checkOpenness();
+
+        if (clusterId == ORID.CLUSTER_ID_INVALID)
+          // GET THE DEFAULT CLUSTER
+          clusterId = defaultClusterId;
+
+        final int finalClusterId = clusterId;
+        return new Iterator<OBrowsePage>() {
+          private OBrowsePage page = null;
+          private long lastPos = 0;
+
+          @Override
+          public boolean hasNext() {
+            if (page == null) {
+              page = nextPage(finalClusterId, lastPos);
+              if (page != null)
+                lastPos = page.getLastPosition();
+            }
+            return page != null;
+          }
+
+          @Override
+          public OBrowsePage next() {
+            if (!hasNext()) {
+              throw new NoSuchElementException();
+            }
+            return next();
+          }
+        };
+      } finally {
+        stateLock.releaseReadLock();
+      }
+    } catch (RuntimeException ee) {
+      throw logAndPrepareForRethrow(ee);
+    } catch (Error ee) {
+      throw logAndPrepareForRethrow(ee);
+    } catch (Throwable t) {
+      throw logAndPrepareForRethrow(t);
+    }
+  }
+
+  private OBrowsePage nextPage(int clusterId, long lastPosition) {
+    try {
+      checkOpenness();
+      stateLock.acquireReadLock();
+      try {
+        checkOpenness();
+
+        final OCluster cluster = doGetAndCheckCluster(clusterId);
+        OPhysicalPosition[] nexts = cluster.higherPositions(new OPhysicalPosition(lastPosition));
```

Comment:
You use higherPositions call, it means that passed in position, will be omitted and also you use lastPosition's default value equal to 0. So during the first call of next method, record with position 0 will not be read. Could you change default value to -1 or explain why that is not the case

Reply:
thanks i'll change that